### PR TITLE
Make the cards-grid filter work

### PR DESCRIPTION
**Pull request by FadhlanR**
The filter list items will be generated from the result of the card type summary API, which provides two key pieces of data: `displayName` and `codeRef`/`id`. The `codeRef` will be used to construct a query for each filter list item, so when the selected item changes, we will re-trigger the pre-rendered search.

However, there are two predefined filter items: `All Cards` and `All Apps`. For `All Cards`, I can reuse the previous implementation of the cards-grid query. For `All Apps`, I need to statically define the `codeRef` of `AppCard` for the query, which is why in this PR, I moved the `AppCard` from the experiment realm to the base realm, as we have a static URL for the base realm.

https://github.com/user-attachments/assets/b03ca015-0085-46d8-ac6e-f137d7132952

### CODE REVIEW

File: packages/base/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "target": "es2020",
     "allowJs": true,
-    "moduleResolution": "node",
+    "moduleResolution": "node16",
```
Comment (jurgenwerk) [inline packages/base/tsconfig.json]: is this relevant to this PR?

File: packages/realm-server/tsconfig.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "inlineSourceMap": true,
     "inlineSources": true,
     "baseUrl": ".",
-    "module": "CommonJS",
+    "module": "nodenext",
```
Comment (jurgenwerk) [inline packages/realm-server/tsconfig.json]: Is this needed for this PR?

**Review by jurgenwerk — APPROVED** [review]
Looks good to me, I tested this locally and works great. I miss counts of items that would tell me how many items are there in each segment (if that's not planned already I think we should consider it).

**Comment by FadhlanR** [comment]
~~Please hold off review for this PR. Based on the discussion in boxel meeting, I'm gonna make a few updates to replace `All Apps` filter with `Favorites` filter.~~

This PR is ready for another review. I've implemented `Favourites` filter as we discussed in boxel meeting.

https://github.com/user-attachments/assets/2864d620-ccee-4488-aa9d-63a3aad9d1ac

File: packages/host/tests/cards/app-card.gts [hunk 3]
```diff
@@ -159,7 +159,7 @@ class AppCardIsolated extends Component<typeof AppCard> {
         {{/if}}
       </div>
     </section>
-    <style scoped>
+    <style>
```
Comment (tintinthong) [inline packages/host/tests/cards/app-card.gts]: This is not scoped?

**Review by tintinthong — APPROVED** [review]

File: packages/base/cards-grid.gts [hunk 4]
```diff
@@ -166,48 +178,71 @@ class Isolated extends Component<typeof CardsGrid> {
...
-  private filters: Filter[] = [
+  constructor(owner: any, args: any) {
+    super(owner, args);
+    this.loadFilterList.perform();
+  }
+
+  filters: { displayName: string; query: any }[] = new TrackedArray([
     {
-      displayName: 'All Apps',
+      displayName: 'Favourites',
```
Comment (jurgenwerk) [inline packages/base/cards-grid.gts]: Since the app is primarily en-US I think this should be Favorites

File: packages/base/cards-grid.gts [hunk 5]
```diff
@@ -269,6 +273,71 @@ class Isolated extends Component<typeof CardsGrid> {
...
+        },
+      });
+    });
+  });
+
+  private get hideCards() {
+    return (
+      this.activeFilter.displayName === 'Favourites' &&
+      (!this.args.model['favourites'] ||
+        this.args.model['favourites'].length === 0)
```
Comment (jurgenwerk) [inline packages/base/cards-grid.gts]: Here you are using `favourites` while the relationship in the model is named `favorites`

**Review by jurgenwerk — COMMENTED** [review]
1. Favorites is empty, is that intentional? Is there a plan to add a couple of default items?


image

File: packages/base/cards-grid.gts (hunk 5 above)
Comment (jurgenwerk) [reply packages/base/cards-grid.gts]: Oh is this why my favorites are not showing?

Comment (FadhlanR) [reply packages/base/cards-grid.gts]: Ah my bad, Fixed it.

**Comment by FadhlanR** [comment]
> 1. Favorites is empty, is that intentional? Is there a plan to add a couple of default items?
> 
> image

We haven't discussed for default favorites items. Based on the previous discussion, the `favorites` filter will be added by the user following the same process used for editing other `linksToMany` fields. Correct me If I'm wrong @lukemelia .

**Comment by lukemelia** [comment]
Our seed realm will provide default favorites for new users' workspaces. Changing favorites will rely on default edit template for now, as Fadhlan described. In the future, we'll add another affordance to favorite/un-favorite a card.

File: packages/base/cards-grid.gts [hunk 6]
```diff
@@ -269,6 +273,71 @@ class Isolated extends Component<typeof CardsGrid> {
...
+              },
+              by: 'title',
+            },
+          ],
+        },
+      });
+    });
+  });
+
+  private get hideCards() {
```
Comment (lukemelia) [inline packages/base/cards-grid.gts]: I think this would be clearer if it was named `isShowingFavorites`

File: packages/base/cards-grid.gts [hunk 7]
```diff
@@ -269,6 +273,71 @@ class Isolated extends Component<typeof CardsGrid> {
...
+    return (
+      this.activeFilter.displayName === 'Favorites' &&
+      (!this.args.model['favorites'] ||
```
Comment (lukemelia) [inline packages/base/cards-grid.gts]: Is there a reason this can't be `this.args.model.favorites` rather than the string index style?

File: packages/host/tests/integration/components/operator-mode-test.gts [hunk 8]
```diff
@@ -2562,6 +2578,40 @@ module('Integration | operator-mode', function (hooks) {
...
+    await click(`[data-test-card-catalog-go-button]`);
+
+    await click('[data-test-edit-button]');
+    assert.dom(`[data-test-cards-grid-item]`).exists({ count: 2 });
+    assert
+      .dom(`[data-test-cards-grid-item="${testRealmURL}Person/1"]`)
+      .exists();
+    assert
+      .dom(`[data-test-cards-grid-item="${testRealmURL}Person/10"]`)
+      .exists();
```
Comment (lukemelia) [inline packages/host/tests/integration/components/operator-mode-test.gts:2616]: Let's assert (here or in another test) that the embedded format is used to render favorites.

**Review by lukemelia — CHANGES_REQUESTED** [review]
